### PR TITLE
(maint) Bump cmake requirement to 3.2.2

### PR DESCRIPTION
**Pull request by MikaelSmith**
Required for Boost 1.5.7, and adds support for specifying a VERSION on
the project macro.

### CODE REVIEW

**Comment by MikaelSmith** [comment]
Projects using this will need to update their cmake_minimum_required and CI scripts when updating the submodule dependency.

File: cmake/cflags.cmake [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Set compiler-specific flags
 # Each of our project dirs sets CMAKE_CXX_FLAGS based on these. We do
 # not set CMAKE_CXX_FLAGS globally because gtest is not warning-clean.
-if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "\\w*Clang")
```
Comment (branan) [inline cmake/cflags.cmake]: This isn't described in the commit message - whaddup here?

Comment (MikaelSmith) [inline cmake/cflags.cmake]: I'll update the message. On Apple the ID changed to AppleClang in a cmake release (not sure which version).

Comment (branan) [inline cmake/cflags.cmake]: I figured it was Apple clang, but wanted to be sure

**Comment by branan** [comment]
@MikaelSmith this conflicts, probably with the boost version bump PR

**Pull request merged**